source/Halibut.Tests/Timeouts/SendingAndReceivingRequestMessagesTimeoutsFixture.cs

```diff
@@ -159,16 +159,33 @@ public async Task WhenThenNetworkIsPaused_WhileSendingARequestMessage_ATcpWriteT
                     addControlMessageTimeout += HalibutLimits.TcpClientHeartbeatSendTimeout;
                 }
 
-                var expectedTimeOut = HalibutLimits.TcpClientSendTimeout // What we actually expected. 
-                                     + HalibutLimits.TcpClientSendTimeout // an extra timeout because of the dispose method of the zip stream (see below)
-                                     + addControlMessageTimeout;
-                sw.Elapsed.Should().BeGreaterThan( expectedTimeOut- TimeSpan.FromSeconds(2), 
-                        "We 'should' wait the send timeout amount of time NOT the heart beat timeout, however when an error occurs writing to the zip (deflate)" +
-                                "stream we also call dispose which again attempts to write to the stream. Thus we wait 2 times the TcpClientSendTimeout.") // -2s give it a little slack to avoid it timed out slightly too early.
-                    .And
-                    .BeLessThan(expectedTimeOut + LowerHalibutLimitsForAllTests.HalfTheTcpReceiveTimeout, 
-                        "We 'should' wait the send timeout amount of time, however when an error occurs writing to the zip (deflate)" +
+                if (clientAndServiceTestCase.ForceClientProxyType == ForceClientProxyType.SyncClient)
```

Comment:
I think the `SyncOrAsync` flag on the test case would be better to use.

FYI, I did a similar change in [my last PR](https://github.com/OctopusDeploy/Halibut/pull/436/files) in this area. It might be worth rebasing and adjusting appropriately.